### PR TITLE
Standard schema

### PR DESCRIPTION
**Pull request by davidrapson**
Adds a minimal schema. Test-drove this so left granular commits here to show the process.

Keen to gradually _pull in_ functionality shared with awards for all rather than starting with it wholesale to give us change to check what abstractions actually make sense.

First step here has been to lift up the `joi-extensions` which highlights the coupling some of them have to awards for all form which can be fixed by pulling in some of the changes proposed originally in https://github.com/biglotteryfund/blf-alpha/pull/2198. Going to do that and then write an initial schema document for this PR.

### CODE REVIEW

**Comment by davidrapson** [comment]
[Define form schema for Hyphen8 to allow them to develop new Forms Data endpoint](https://trello.com/c/HNMQKc9d/771-define-form-schema-for-hyphen8-to-allow-them-to-develop-new-forms-data-endpoint)

**Comment by davidrapson** [comment]
Got this to the stage where I have:

- An initial schema definition
- A set of unit tests for the form schema
- Initial documentation
- Decoupled our `joi-extensions` from the awards for all implementation

I think that's enough for this PR, so this is ready for review now.

**Comment by NikhilNanjappa** [comment]
Awesome stuff 👍

**Comment by davidrapson** [comment]
@mattandrews @NikhilNanjappa Unlikely to merge this before I leave for the week so once you are both happy feel free to merge this next week. Might be worth doing a "squash and merge" on this one to keep the commit log smaller.

File: controllers/apply/get-advice/schema.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        .required(),
+    contactName: Joi.fullName().required(),
+    contactEmail: Joi.string()
+        .email()
+        .required(),
+    contactPhone: Joi.string()
+        .phoneNumber()
+        .allow('')
+        .optional()
+});
```
Comment (mattandrews) [inline controllers/apply/get-advice/schema.js]: 💯

File: controllers/apply/lib/joi-extensions/full-name.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+module.exports = function fullName(joi) {
+    return {
+        name: 'fullName',
+        base: joi.object({
+            firstName: joi
+                .string()
+                .max(40)
```
Comment (mattandrews) [inline controllers/apply/lib/joi-extensions/full-name.js]: Out of interest, why the different maxlength between first/last name? Might as well make them the same for both (cf. https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/).

Comment (mattandrews) [reply controllers/apply/lib/joi-extensions/full-name.js]: (or is this a Salesforce thing?)

Comment (davidrapson) [reply controllers/apply/lib/joi-extensions/full-name.js]: This is a Salesforce limit sadly.

Comment (mattandrews) [reply controllers/apply/lib/joi-extensions/full-name.js]: I'm sad that my future child Llanfairpwllgwyngyllgogerychwyrndrobwllllantysiliogogogoch (named for my newly-discovered love of the Welsh language) will be unable to apply for funding.

**Pull request merged**